packages/pigeon/lib/linux_generator.dart

```diff
@@ -0,0 +1,1309 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'ast.dart';
+import 'generator.dart';
+import 'generator_tools.dart';
+
+/// General comment opening token.
+const String _commentPrefix = '//';
+
+/// Documentation comment spec.
+const DocumentCommentSpecification _docCommentSpec =
+    DocumentCommentSpecification(_commentPrefix);
+
+/// Options that control how Linux code will be generated.
+class LinuxOptions {
+  /// Creates a [LinuxOptions] object
+  const LinuxOptions({
+    this.headerIncludePath,
+    this.module = 'My',
```

Comment:
I missed this until I tried using it; I don't think we want a random string as the default. It should either be nothing, or be something meaningful like the package name converted from snake case.

Reply:
For GObject we have to use something for the module name. I chose `My` as this matches the Linux template. It's not great though, and I'm open to ideas. As per the previous comment perhaps we should just force the user to choose something, which is a small hurdle to get started but avoids the weird default.

Reply:
I think the package name is fine. I'm also open to having this as a required field.

Reply:
Now using the package name by default.